Vagrantfile for PowerFit-DisVis VM and required modules/assets

oyzeyno opened the pull request:
Vagrantfile.powerfit_disvis: For creating a VM based on courseVM+Chimera, using following modules:
module_base.sh: same as "main.sh" in molmod, python lib lines are deleted
module_tools.sh: haddock-tools and pdb-tools
module_python2.7.sh: python2.7.10and pip2.7 installation
module_powerfit_disvis.sh: powerfit, disvis installation with required libraries (including numpy scipy pyfftw etc)

oyzeyno committed:
Vagrantfile for PowerFit-DisVis VM and required modules/assets

JoaoRodrigues commented on provisioning/module_base.sh:
```diff
@@ -0,0 +1,43 @@
+
+# Building tools
+apt-get -qq install -y --no-install-recommends git autoconf automake cmake autotools-dev build-essential > /dev/null
+apt-get install curl
```
`apt-get` is really verbose, so I'm silencing most of its regular output (not errors!) with `-qq` and a redirection to `/dev/null`. Also, to avoid unnecessary things being installed, and to make sure all prompts are answered _yes_ (sometimes it asks you if you want to update, install, etc), I added the `--no-install-recommends` and `-y`flags. See the previous line. My suggestion is that all apt-get commands should follow this pattern, unless there is a specific need.

oyzeyno replied:
Ok, I will revise the script accordingly

JoaoRodrigues commented on vagrantfiles/Vagrantfile.powerfit_disvis:
```diff
@@ -0,0 +1,81 @@
+                   "--vram", "128",
+                   "--paravirtprovider", "kvm" # for Linux Guests 
+    ]
+    vb.customize ["modifyvm", :id, "--accelerate3d", "on"]
```
This might break Pymol. Actually, I'm almost sure this will break Pymol :) Is this necessary for Chimera?

oyzeyno replied:
I read your note about "No 3D acceleration!" on molmod page before, but Chimera needs 3D acceleration. It craves 3D acceleration. So, I will leave this on, and so far it's been more or less stable.

JoaoRodrigues commented on vagrantfiles/Vagrantfile.powerfit_disvis:
```diff
@@ -0,0 +1,81 @@
+    vb.customize ["modifyvm", :id, "--accelerate3d", "on"]
+
+    # Customize number of CPUs
+    vb.cpus = 4
```
Not everyone has such a nice computer!! We should leave the default at 1. People can change it when they run it themselves.

oyzeyno replied:
The reason for "4" CPUs was to trying powerfit and disvis on multiple CPU, but I will change it to 1.

JoaoRodrigues commented on provisioning/module_python2.7.sh:
```diff
@@ -0,0 +1,21 @@
+
+# Install dependencies for PowerFit and DisVis
+echo "[++] Downloading & Installing Python 2.7.10"
+apt-get install build-essential checkinstall
```
`build-essential` is already in `module_main`.

oyzeyno replied:
Ok, I deleted it

JoaoRodrigues commented on provisioning/module_python2.7.sh:
```diff
@@ -0,0 +1,21 @@
+
+echo "[++] Downloading pip for Python2.7.10"
+curl -O https://bootstrap.pypa.io/get-pip.py
+python2.7 /opt/software/Python-2.7.10/get-pip.py
```
I'm adding the comment here but it relates to the entire script. See how I enclose the commands in parenthesis in other modules, and redirect the output to `/dev/null`. Otherwise this will output a **ton** to stdout, namely the `./configure` and `make`.

oyzeyno replied:
Yeah, I saw that and used that structuring in powerfit and disvis installation from github in module_powerfit_disvis.sh. I revised python2.7 module accordingly.

JoaoRodrigues commented on provisioning/module_powerfit_disvis.sh:
```diff
@@ -0,0 +1,36 @@
+apt-get install	-y libfftw3-dev cython > /dev/null
+apt-get install -y python-dev > /dev/null
+pip2.7 -q install --upgrade pip > /dev/null
+pip2.7 -q install --upgrade virtualenv > /dev/null
```
I saw this in the previous script, but where is virtualenv used? This is very likely only for development.

JoaoRodrigues commented:
Thanks @oyzeyno !

A few comments: 
- I don't think we need a `module_base` and a `module_main` if 95% of the file is the same. We can just split `module_main` or add things wherever necessary. It's OK to have a couple of extra things here and there (e.g. curl) that might not be necessarily needed by all machines, as long as they are simple to install and small.
- The same as above for haddock-tools. We should just add it to `module_generic`.
- `module_python` is never called by your provisioner.

See if you agree with my comments in the scripts themselves.

Do not merge!

oyzeyno committed:
modified:   assets/molmod/config/bashrc
Two more aliases are added.
	modified:   provisioning/module_generic.sh
haddock-tools installation is added Biopython is commented out
	modified:   vagrantfiles/Vagrantfile.powerfit_disvis
vb.name is added, provisioning modules are rearranged

oyzeyno committed:
Typo in Vagrantfile

oyzeyno committed:
make-contact-histogram.csh is linked to /opt/bin
